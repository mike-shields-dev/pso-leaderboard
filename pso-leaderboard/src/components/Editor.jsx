import React from "react";
import { Link } from "react-router-dom";
import { HelmetProvider, Helmet } from "react-helmet-async";

const Editor = () => {
  return (
    <div>
      <HelmetProvider>
        <Helmet>
          <title>Penalty Shootout: Editor</title>
        </Helmet>
      </HelmetProvider>
      <p>Editor</p>
      <Link to="/">Home</Link>
      <Link to="/display">Display</Link>
    </div>
  );
};

export default Editor;
