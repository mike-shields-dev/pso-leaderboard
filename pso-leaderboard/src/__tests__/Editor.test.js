import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Editor from "../components/Editor";

const renderEditor = () => render(<Editor />, { wrapper: BrowserRouter });

describe("Editor", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderEditor();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders hyperlinks to '/home' and '/display'", () => {
    renderEditor();

    expect(screen.getByRole("link", { name: /home/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /display/i })).toBeInTheDocument();
  });
});
