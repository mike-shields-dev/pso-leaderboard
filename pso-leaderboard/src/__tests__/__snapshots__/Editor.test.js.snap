// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      Editor
    </p>
    <a
      href="/"
    >
      Home
    </a>
    <a
      href="/display"
    >
      Display
    </a>
  </div>
</DocumentFragment>
`;
